src/protocol.cpp

```diff
@@ -125,18 +123,6 @@ bool CMessageHeader::IsCommandValid() const
     return true;
 }
 
-
-ServiceFlags GetDesirableServiceFlags(ServiceFlags services) {
-    if ((services & NODE_NETWORK_LIMITED) && g_initial_block_download_completed) {
```

Comment:
Commit a710e08398 `net: move state dependent peer services flags` moves this code to `src/net_processing.cpp` and changes from `g_initial_block_download_completed` to `m_initial_sync_finished`. The variable `m_initial_sync_finished` is used for something else and has a different semantic, thus using `m_initial_sync_finished` here is wrong. Later another commit changes from `m_initial_sync_finished` to the proper check.

I think it would be better to reorder the commits, so that the proper check (either `m_allow_limited_peers_conn` or `ApproximateBestBlockAgeInNumberOfBlocks()` as I suggested in another comment) is already available in peerman and so when `GetDesirableServiceFlags()` is moved, it switches the proper check right away.

Reply:
Sure. Have reordered the commits.